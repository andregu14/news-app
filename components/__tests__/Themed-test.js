import { render, screen } from '@testing-library/react-native';
import { Text, View } from '../Themed';

describe('<Themed />', () => {
    it('renders ThemedText correctly', () => {
        render(<Text>Test Text</Text>);
        expect(screen.getByText('Test Text')).toBeVisible();
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders ThemedView correctly', () => {
        render(<View><Text>Child Text</Text></View>);
        expect(screen.getByText('Child Text')).toBeVisible();
        expect(screen.toJSON()).toMatchSnapshot();
    });
});
