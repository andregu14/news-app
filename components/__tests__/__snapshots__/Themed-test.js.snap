// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Themed /> renders ThemedText correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
        "fontFamily": "Inter_400Regular",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`<Themed /> renders ThemedView correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
          "fontFamily": "Inter_400Regular",
        },
        undefined,
      ]
    }
  >
    Child Text
  </Text>
</View>
`;
